### PR TITLE
Improve copy to clipboard code

### LINKED ISSUE #18529: add copy to export like text
In the export section, there is an option that indicates if I want this in a text box, will it be possible to add a copy button to the result view??? currently it says "update" and "back"

image

### PR DESCRIPTION
**Pull request by kamioon**
### Description

This is a PR for adding a **copy** to the clipboard on the export page that was requested (#18529). 

phpmyadmin_copy_export

To keep the text format copied to the clipboard temporary `input` has changed to the `textarea`.

### CODE REVIEW

**Review by williamdes — APPROVED** [review]
Very nice

File: js/src/modules/functions.ts [hunk 1]
```diff
@@ -1516,8 +1516,13 @@ function dismissNotifications () {
 
         $(document).on('click', 'a.copyQueryBtn', function (event) {
             event.preventDefault();
-            var res = Functions.copyToClipboard($(this).attr('data-text'));
-            if (res) {
+            var copyText = $(this).attr('data-text');
+            var copyTarget = $(this).attr('data-target');
+            if (copyTarget) {
+                copyText = String($(copyTarget).val());
```
Comment (williamdes) [inline js/src/modules/functions.ts:1522]:
```suggestion
                copyText = $(copyTarget).val();
```

**Review by williamdes — APPROVED** [review]
💯

**Comment by williamdes** [comment]
Can you check why tests are not passing?

**Review by williamdes — CHANGES_REQUESTED** [review]
what was wrong with the previous version?

**Comment by kamioon** [comment]
`[tsl] ERROR in /home/runner/work/phpmyadmin/phpmyadmin/js/src/modules/functions.ts(1522,17)
      TS2322: Type 'string | number | string[]' is not assignable to type 'string'.`
@williamdes the `.val()` returns a mixed type and that was the reason I had added the `String()`. I've changed a little bit more. separated them into two different functions and moved the copy message to a new function. I think it's better now. otherwise, reverting the two latest commits could work fine but your commit suggestion wouldn't be applied there.

**Comment by kamioon** [comment]
> what was wrong with the previous version?

Sorry, I just saw that you had approved the previous one. I explained here the reason for failing the test and I think the new changes are cleaner than using the `copyQueryBtn` for two purposes. (it should handle two different text sources.)

> `[tsl] ERROR in /home/runner/work/phpmyadmin/phpmyadmin/js/src/modules/functions.ts(1522,17) TS2322: Type 'string | number | string[]' is not assignable to type 'string'.` @williamdes the `.val()` returns a mixed type and that was the reason I had added the `String()`. I've changed a little bit more. separated them into two different functions and moved the copy message to a new function. I think it's better now. otherwise, reverting the two latest commits could work fine but your commit suggestion wouldn't be applied there.

File: js/src/modules/functions.ts [hunk 2]
```diff
@@ -1516,17 +1516,28 @@ function dismissNotifications () {
...
+            displayCopyStatus(this, copyStatus);
+        });
+
+        $(document).on('click', 'a.copyExportBtn', function (event) {
+            event.preventDefault();
+            var copyTarget = $(this).attr('data-target');
+            var copyStatus = Functions.copyToClipboard($(copyTarget).val());
+            displayCopyStatus(this, copyStatus);
+        });
+
```
Comment (williamdes) [inline js/src/modules/functions.ts:1529]: could you add a block of documentation like other functions to document the input and output
or much better: add typescript types

**Review by williamdes — APPROVED** [review]
Thank for explaining, looks okay
I left one minor comment about typing

**Review by williamdes — APPROVED** [review]
This really looks good

**Pull request merged**